cpp/src/arrow/util/cpu_info.cc

```diff
@@ -183,14 +183,13 @@ bool RetrieveCPUInfo(int64_t* hardware_flags, std::string* model_name) {
 CpuInfo::CpuInfo() : hardware_flags_(0), num_cores_(1), model_name_("unknown") {}
 
 std::unique_ptr<CpuInfo> g_cpu_info;
-static std::mutex cpuinfo_mutex;
+static std::once_flag cpuinfo_once;
```

Comment:
How about using `cpuinfo_initialized` instead of `cpuinfo_once`?
We use `XXX_initialized` naming in `cpp/src/arrow/util/utf8.cc` and `cpp/src/arrow/extension_type.cc`.

Reply:
done